packages/cloud-cognitive/src/components/StatusIcon/StatusIcon.js

```diff
@@ -120,46 +120,54 @@ const icons = {
   },
 };
 
-const blockClass = `${pkg.prefix}-status-icon`;
+const blockClass = `${pkg.prefix}--status-icon`;
+const componentName = 'StatusIcon';
 
-export const StatusIcon = ({ type, theme, size, className, ...rest }) => {
-  const [icon, setIcon] = useState([]);
-  const [iconSize, setIconSize] = useState('');
-  const [iconTheme, setIconTheme] = useState(null);
-  const IconComponent = icon[iconSize];
-
-  const classNames = cx({
-    [`${blockClass}--${iconTheme}`]: iconTheme,
-    [`${blockClass}--${iconTheme}-${type}`]: type,
-    [className]: className,
-  });
+const isValidIconDescription = () => (props) => {
+  if (props.iconDescription.toLowerCase() != props.kind) {
+    throw new Error(
+      'The "icon description" prop must match the status icon "kind" prop'
+    );
+  } else return;
+};
```

Comment:
If both are these are to be the same, is there a reason why i would have to pass down both? Or can you generate the iconDescription from the kind by capitalizing the first word?

Reply:
You can also do this in the prop types which I think will make sense because you're validating the props.

<img width="597" alt="Screen Shot 2021-04-09 at 9 34 57 AM" src="https://user-images.githubusercontent.com/12755042/114188186-e4176200-9916-11eb-8a65-0d870d99dfec.png">

Reply:
In fact, I don't think we can link these at all: `iconDescription` needs to be translatable, so we can't generate it from the kind. We could perhaps generate a default icon description from the kind.

Reply:
Yeah because here it's indicating that the kind and iconDescription have to be the same, what if we didnt? I really don't know the design decision but was just wondering.